ea-2022-04-ndvi-automation.py

```diff
@@ -0,0 +1,901 @@
+    Band 3 = Green -> Index 1
+    Band 4 = Red -> Index 2
+    Band 5 = NIR -> Index 3
+
+    Parameters
+    ----------
+    files: List
+        A list of Landsat tif images
+    crop_boundary : shapefile dataArray
+
+    Returns
+    -------
+    data_xr : dataArray
+        Consolidated all bands in files cropped to shapefile
+
+    """
+    all_bands = []
+    for i, aband in enumerate(files):
+        all_bands.append(
+            (rxr.open_rasterio(aband, masked=True)
+             # Inlcuding from_disk=True makes code run faster
+             .rio.clip(crop_boundary.geometry, from_disk=True)
+             .squeeze())
+        )
+        # Assign a band number to the new xarray object
+        all_bands[i]["band"] = i+1
+
+    # Turn list of bands into a single xarray object
+    data_xr_crop = xr.concat(all_bands, dim="band")
+
+    # IMPORTANT: Clean the dataArray - remove 0's and high values
+    data_nozeros_xr = data_xr_crop.where(data_xr_crop > 0, np.nan)
+    data_nozeros_xr = data_nozeros_xr.where(data_nozeros_xr < 10000, np.nan)
+
+    return data_nozeros_xr
+
+
+# ### Function 4 - Apply cloud mask
+
+# In[10]:
+
+
+def apply_cloud_mask(data_xr, qa_file, crop_boundary):
+    """Applies a cloud mask to Landsat dataArray
+
+    Parameters
+    ----------
+    qa_file : String
+        Path to the Landsat quality flag file
+
+    data_xr : dataArray
+        Landsat 8 band consolidated dataArray
+
+    crop_boundary : shapefile dataArray
+
+    Returns
+    -------
+    Cloud free and cropped dataArray
+
+    """
+    # Read the quality flags file
+    qa_xr = (rxr.open_rasterio(qa_file, masked=True)
+             .rio.clip(crop_boundary.geometry, from_disk=True)
+             .squeeze())
+
+    # Create cloud mask using earthpy mask package for Landsat imagery
+    high_cloud_confidence = (
+        em.pixel_flags["pixel_qa"]["L8"]["High Cloud Confidence"])
+    cloud = em.pixel_flags["pixel_qa"]["L8"]["Cloud"]
+    cloud_shadow = em.pixel_flags["pixel_qa"]["L8"]["Cloud Shadow"]
+
+    # Add up all the mask values
+    all_masked_values = cloud_shadow + cloud + high_cloud_confidence
+
+    # Apply the masking to the data xarray
+    data_cld_free_xr = data_xr.where(~qa_xr.isin(all_masked_values))
+
+    return data_cld_free_xr
+
+
+# ### Function 5 - Calculate NDVI and return the mean
+
+# In[11]:
+
+
+def calc_mean_ndvi(data_xr):
+    """Calculates the NDVI and returns the mean
+
+    Reference: https://stackoverflow.com/questions/49867345/
+    how-to-deal-with-inf-values-when-computting-the-average-of-values-of-a-list-in-p
+
+    Parameters
+    ----------
+    data_xr : dataArray
+        The Landsat data
+
+    Returns
+    -------
+    float : numpy
+        The mean NDVI
+    """
+    ndvi_xr = es.normalized_diff(data_xr[3], data_xr[2])
+
+    # Calculating my own NDVI
+    # ndvi_xr = (data_xr[3] - data_xr[2]) / (data_xr[3] + data_xr[2])
+    # ndvi_clean = ndvi_xr.where(np.isfinite(ndvi_xr.values))
```

Comment:
No harm in leaving this (I think I have a ton of commented-out-scratch-work in my code), but your code is so concise and efficient that leaving this in stuck out to me. Again, very minor!